packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -552,6 +555,31 @@ mixin WidgetsBinding on BindingBase, ServicesBinding, SchedulerBinding, GestureB
         },
       );
 
+      registerServiceExtension(name: 'invokePreHotRestartCallbacks', callback: (Map<String, Object> params) async {
```

Comment:
This should probably be in the BaseBinding or ServicesBinding, rather than WidgetsBinding, since it's not related to widgets. Probably BaseBinding (in foundation).

Reply:
I moved it to BindingBase, but since WidgetsBinding is mixing all bindings, I'm assuming the user would still be using it as `WidgetsBinding.instance.debugRegisterHotRestartCallback`. Is that correct?